src/qt/ve_qitems_mqtt.cpp

```diff
@@ -0,0 +1,477 @@
+#include <veutil/qt/ve_qitems_mqtt.hpp>
+
+#include <QRandomGenerator>
+#include <QJsonDocument>
+#include <QJsonObject>
+
+//--
+
+VeQItemMqtt::VeQItemMqtt(VeQItemMqttProducer *producer)
+	: VeQItem(producer, nullptr)
+{
+	connect(producer, &VeQItemMqttProducer::connectionStateChanged,
+		this, [this] {
+			const VeQItemMqttProducer::ConnectionState state = mqttProducer()->connectionState();
+			switch (state) {
+			case VeQItemMqttProducer::Ready:
+				// TODO: may not be required, since we subscribe to everything
+				//       and send keepalive to read all values when Ready.
+				mqttProducer()->readValue(uniqueId());
+				break;
+			case VeQItemMqttProducer::Disconnected:
+				produceValue(QVariant(), VeQItem::Offline);
+				break;
+			default: break;
+			}
+		});
+}
+
+int VeQItemMqtt::setValue(QVariant const &value)
+{
+	if (!mqttProducer() || !mqttProducer()->publishValue(uniqueId(), value)) {
+		return -1;
+	}
+	return 0;
+}
+
+// This is called by VeQItem::itemAddChild().
+// TODO: may not be necessary, remove?
+void VeQItemMqtt::setParent(QObject *parent)
+{
+	VeQItem::setParent(parent);
+}
+
+VeQItemMqttProducer *VeQItemMqtt::mqttProducer() const
+{
+	return static_cast<VeQItemMqttProducer*>(mProducer);
+}
+
+//--
+
+VeQItemMqttProducer::VeQItemMqttProducer(
+		VeQItem *root, const QString &id, const QString &clientIdPrefix, QObject *parent)
```

Comment:
formatting?

Reply:
It wasn't modified in the last commit so I'm hesitant to amend it with a change here, as it's unrelated.